packages/sdk/src/protocols/protocol.ts

```diff
@@ -198,9 +198,14 @@ export class Protocol extends EventEmitter {
 
     protected handleReplyMessage(message: Message) {
         const queuedPromise = this.promises[message.replyToId];
+        // The way join-synchronization works, the app may recieve the occasional unexpected reply message.
+        // It is harmless; something we can live with for now. We do want to know if we reeive an unexpected reply
+        // elsewhere, as that could be a sign of a protocol design issue.
         if (!queuedPromise) {
-            // tslint:disable-next-line:no-console max-line-length
-            console.error(`[ERROR] ${this.name} received unexpected reply message! payload: ${message.payload.type}, replyToId: ${message.replyToId}`);
+            if (this.name !== "Execution") {
+                // tslint:disable-next-line:no-console max-line-length
+                console.error(`[ERROR] ${this.name} received unexpected reply message! payload: ${message.payload.type}, replyToId: ${message.replyToId}`);

```

Comment:
Is there a reason that this is console.error and not through our normal logging mechanism?

Reply:
Until very recently we didn't have app logging enabled by default on our samples and functional tests, and I really want to know if this condition was ever hit, so I had it go through the console. It could be switched over to app logging now. There are a few other places in the code where we're logging to console. I'll clean them up in a separate PR.